examples/custom-congestion-controller/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+            sent_bytes: usize,
+            app_limited: Option<bool>,
+            rtt_estimator: &RttEstimator,
+            publisher: &mut Pub,
+        ) -> Self::PacketInfo {
+            self.bytes_in_flight += sent_bytes as u32;
+        }
+
+        fn on_rtt_update<Pub: Publisher>(
+            &mut self,
+            time_sent: Timestamp,
+            now: Timestamp,
+            rtt_estimator: &RttEstimator,
+            publisher: &mut Pub,
+        ) {
+            // no op
+        }
+
+        fn on_ack<Pub: Publisher>(
+            &mut self,
+            newest_acked_time_sent: Timestamp,
+            bytes_acknowledged: usize,
+            newest_acked_packet_info: Self::PacketInfo,
+            rtt_estimator: &RttEstimator,
+            random_generator: &mut dyn RandomGenerator,
+            ack_receive_time: Timestamp,
+            publisher: &mut Pub,
+        ) {
+            self.bytes_in_flight -= bytes_acknowledged as u32;
+            self.congestion_window += bytes_acknowledged as u32;
+        }
+
+        fn on_packet_lost<Pub: Publisher>(
+            &mut self,
+            lost_bytes: u32,
+            packet_info: Self::PacketInfo,
+            persistent_congestion: bool,
+            new_loss_burst: bool,
+            random_generator: &mut dyn RandomGenerator,
+            timestamp: Timestamp,
+            publisher: &mut Pub,
+        ) {
+            self.bytes_in_flight -= lost_bytes;
+            self.congestion_window = (self.congestion_window as f32 * 0.5) as u32;
+        }
+
+        fn on_explicit_congestion<Pub: Publisher>(
+            &mut self,
+            ce_count: u64,
+            event_time: Timestamp,
+            publisher: &mut Pub,
+        ) {
+            self.congestion_window = (self.congestion_window as f32 * 0.5) as u32;
+        }
+
+        fn on_mtu_update<Pub: Publisher>(&mut self, max_data_size: u16, publisher: &mut Pub) {
+            // no op
+        }
+
+        fn on_packet_discarded<Pub: Publisher>(&mut self, bytes_sent: usize, publisher: &mut Pub) {
+            self.bytes_in_flight -= bytes_sent as u32;
+        }
+
+        fn earliest_departure_time(&self) -> Option<Timestamp> {
+            None
+        }
+    }
+
+    // Define an endpoint for the custom congestion controller so it may be used as a
+    // congestion controller provider to the s2n-quic server or client
+    #[derive(Debug, Default)]
+    pub struct MyCongestionControllerEndpoint {}
+
+    impl congestion_controller::Endpoint for MyCongestionControllerEndpoint {
+        type CongestionController = MyCongestionController;
+
+        // This method will be called whenever a new congestion controller instance is needed.
+        fn new_congestion_controller(
+            &mut self,
+            path_info: congestion_controller::PathInfo,
+        ) -> Self::CongestionController {
+            MyCongestionController {
+                congestion_window: path_info.max_datagram_size as u32,
```

Comment:
Probably not a big deal since this is a toy example but this essentially means you can only send 1 packet per RTT, right? Maybe we multiply it by 10, similar to cubic?